### PR TITLE
Ingested Gspread Data Into Postgres Database and Utitlized Dbeaver to write SQL Scripts

### PR DESCRIPTION
**Pull request by Humphery7**
@elseagle 
@tobaadesugba 
@AmazingGrace-D 

-> Created Postgres Database
-> Utilized gspread library to get data from google spreadsheet  
-> Utilized psycopg2 library to send sql queries to database to perform actions(create_table, truncate_table, insert_into_table...)
-> Installed Dbeaver and connection setup to write sql scripts to database
-> Managed Packages with poetry
-> Validated Incoming data
-> logged errors encounted with logging library

### CODE REVIEW

File: Project5/project5/app.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                'wind_direction': str(record['wind_direction']),
+                'temp_c': float(record['temp_c']),
+                'wind_kph': float(record['wind_kph']),
+                'latitude': float(record['latitude']),
+                'longitude': float(record['longitude']),
+                'timestamp': record['timestamp'] 
+            }
+            validated_data.append(valid_record)
+        except (ValueError, KeyError) as e:
+            print(f"Skipping row due to error: {e}")
```
Comment (elseagle) [inline Project5/project5/app.py]: Do not use print, use the logging library

File: Project5/project5/app.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                record['longitude'],
+                record['timestamp'] if record['timestamp'] else None
+            )
+            with open('Project5/project5/insert_values.sql', 'r') as file:
+                insert_values_sql = file.read()
+            cur.execute(
+                insert_values_sql, values
+            )
+        conn.commit()
+        print("Data inserted successfully")
```
Comment (elseagle) [inline Project5/project5/app.py]: Kindly remove all prints and use logging if needed

File: Project5/project5/app.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                record['state'],
+                record['country'],
+                record['wind_direction'],
+                record['temp_c'],
+                record['wind_kph'], 
+                record['latitude'],
+                record['longitude'],
+                record['timestamp'] if record['timestamp'] else None
+            )
+            with open('Project5/project5/insert_values.sql', 'r') as file:
```
Comment (elseagle) [inline Project5/project5/app.py]: This belongs to .env

File: Project5/project5/config.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from configparser import ConfigParser
+
+
+def config(filename='Project5/project5/database.ini', section='postgresql'):
```
Comment (elseagle) [inline Project5/project5/config.py]: Magic string here should be an environment variable

File: Project5/project5/create_table.sql [hunk 5]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE Weather(
+    id SERIAL PRIMARY KEY,
+    location VARCHAR(100) NOT NULL,
+    state VARCHAR(100) NOT NULL,
+    country VARCHAR(100) NOT NULL,
+    wind_direction VARCHAR(100) NOT NULL,
+    temp_c INT NOT NULL,
```
Comment (elseagle) [inline Project5/project5/create_table.sql]: What's temp_c? Not quite descriptive

File: Project5/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+[tool.poetry]
+name = "project5"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (elseagle) [inline Project5/pyproject.toml]: Your names should be here ¯\_(ツ)_/¯

File: Project5_II/python_sql.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def database_connect_execute():
+    """connects to postgres database
+    and executes sql commands to perform 
+    operations"""
+    #get password from environment variable
+    PASSWORD = os.getenv("PASSWORD")
+    #connect to postgres database
+    with psycopg2.connect(host="localhost",dbname="GspreadData_Storage",
```
Comment (elseagle) [inline Project5_II/python_sql.py]: Your db name needs to look better. You can stick with lower case and use a descriptive name - `gspreaddata_storage` doesn't sound good. You should name your db based on the data, use case or task (I mean what is being achieved)

File: Project5_II/python_sql.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                        user="postgres",password=PASSWORD,port=5432) as conn:
+
+    # Open a cursor to perform database operations
+        with conn.cursor() as cur:
+
+            #sql function to crreate table if it doesn't exist
+            createTable = """CREATE TABLE IF NOT EXISTS Github_Records(
+                ID SERIAL NOT NULL,
```
Comment (elseagle) [inline Project5_II/python_sql.py]: Not a good naming, stick to lower case

File: Project5_II/python_sql.py [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                Repository_Name VARCHAR(100) NOT NULL,
+                Language_Used VARCHAR(50),
+                Description VARCHAR(200),
+                Datetime_Posted VARCHAR(50) NOT NULL,
```
Comment (elseagle) [inline Project5_II/python_sql.py]: `created_at` sounds better.

**Review by elseagle — CHANGES_REQUESTED** [review]
Good work. Thank you.

I noticed you only have 1 PR, the instruction says otherwise. Kindly ensure you give more attention to details in these scenarios. All details matter.
Screenshot 2023-11-25 at 15 36 10

Kindly attend to requested changes too

**Review by Humphery7 — COMMENTED** [review]
@elseagle 
@tobaadesugba 

-> created better names for database and table 
-> names are ensured to be lower case
-> created entry point, script now runs on command (poetry run app)

File: Project5/project5/create_table.sql (hunk 5 above)
Comment (elseagle) [reply Project5/project5/create_table.sql]: This is yet to be resolved @Humphery7 @LogicAL007

**Review by elseagle — APPROVED** [review]

**Pull request merged**